File: retrofit_demo/src/main/java/com/yeqiu/retrofit/utils/LogUtils.java
package com.yeqiu.retrofit.utils;

import com.google.gson.Gson;

/**
 * @project：xinhe-signcenter
 * @author：小卷子
 * @date 2020-02-24
 * @describe：
 * @fix：
 */

public class LogUtils {


    public static void log(String tag, Object msg) {

        if (msg == null) {
            System.out.println("======= " + tag + "= " + null);
        } else if (msg instanceof String) {
            System.out.println("======= " + tag + "= " + msg);
        } else {

            try {
                System.out.println("======= " + tag + "= " + new Gson().toJson(msg));
            } catch (Exception e) {

            }
        }
    }


}

File: AwesomeWeather/HailHydra/src/main/java/com/yeqiu/hailhydra/common/util/LogUtils.kt
package com.yeqiu.hailhydra.common.util

import com.google.gson.Gson
import com.orhanobut.logger.AndroidLogAdapter
import com.orhanobut.logger.FormatStrategy
import com.orhanobut.logger.Logger
import com.orhanobut.logger.PrettyFormatStrategy


/**
 * @project：HailHydra
 * @author：小卷子
 * @date 2020/9/24
 * @describe：
 * @fix：
 */


fun log(msg: Any,tag:String = "") {

    if (msg is String) {
        Logger.t(tag).i(msg)
    } else {
        //转json处理
        val json = Gson().toJson(msg)
        Logger.t(tag).i(json)
    }
}


fun logJson(json: String) {
    Logger.json(json)
}


object LogUtils {

    private val tag = APPUtil.getAppName()

    fun init() {
        val formatStrategy: FormatStrategy = PrettyFormatStrategy.newBuilder()
            .showThreadInfo(true)
            .methodCount(5)
            .methodOffset(0)
            .tag(tag)
            .build()


        Logger.addLogAdapter(AndroidLogAdapter(formatStrategy))
    }

}
